### PR TITLE
add info if product is backordered on product page

### LINKED ISSUE #7752: No information whether the item is backordered or not
There is no information about the item being backordered. Only visible On Admin's panel side and "Orders" in My Account, after the order as conducted.

### PR DESCRIPTION
**Pull request by bbonislawski**
This needs a little bit copy and some feedback if it's ok to show confirm modal when product is backordered
fixes #7752

### CODE REVIEW

File: core/app/models/spree/variant.rb [hunk 1]
```diff
@@ -253,6 +253,10 @@ def discontinued?
       !!discontinue_on && discontinue_on <= Time.current
     end
 
+    def backordered?
+      total_on_hand <= 0 && stock_items.where(backorderable: true).any?
```
Comment (nnande) [inline core/app/models/spree/variant.rb]: @bbonislawski  `&& stock_items.exists?(backorderable: true)` - How about it?

Comment (bbonislawski) [reply core/app/models/spree/variant.rb]: 👍

File: frontend/app/assets/javascripts/spree/frontend/product.js [hunk 2]
```diff
@@ -1,5 +1,10 @@
 Spree.ready(function ($) {
 
+  Spree.toggleBackorderedInfo = function(variant) {
```
Comment (bbonislawski) [inline frontend/app/assets/javascripts/spree/frontend/product.js]: remove this

File: frontend/app/assets/javascripts/spree/frontend/product.js [hunk 3]
```diff
@@ -63,13 +68,24 @@ Spree.ready(function ($) {
...
       Spree.updateVariantPrice($(this));
+      Spree.toggleBackorderedInfo ($ this)
       return Spree.disableCartForm($(this));
     });
   }
 
-  return Spree.addImageHandlers();
+  Spree.addImageHandlers();
+
+  $('#add-to-cart-button').click(function(event) {
```
Comment (bbonislawski) [inline frontend/app/assets/javascripts/spree/frontend/product.js]: remove this

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**